src/base/misc/fatfs.c

```diff
@@ -376,37 +376,35 @@ int read_data(fatfs_t *f, unsigned pos, unsigned char *buf)
 
 static void set_geometry(fatfs_t *f, unsigned char *b)
 {
+  struct on_disk_bpb *bpb = (void *) &b[0x0b];
```

Comment:
Oh... please avoid such things in the
future (I guess I can commit this anyway
if nothing else is found). Use proper casts.